lib/parser/scope.js

```diff
@@ -180,11 +198,25 @@ Scope.prototype.assertIndentifiersExist = function(node) {
   }
 };
 
+Scope.prototype._inferLogicalExpression = function(node) {
+  var left = this.inferType(node.left),
+    right = this.inferType(node.right);
+
+  if (left !== 'boolean') {
+    throw new nodeError(node, `Left operand of ${node.operator} must be boolean.`);
+  }
+
+  if (right !== 'boolean') {
+    throw new nodeError(node, `Right operand of ${node.operator} must be boolean.`);
+  }
```

Comment:
Firebase wants a boolean: `true && auth.isSomething` would be rejected; `true && auth.isSomething == true` would be accepted.